sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs

```diff
@@ -216,5 +218,12 @@ private static TokenCredential[] GetDefaultAzureCredentialChain(DefaultAzureCred
 
             return chain;
         }
+
+        private static DefaultAzureCredentialOptions ValidateAuthorityHostOption(DefaultAzureCredentialOptions options)
+        {
+            Validations.ValidateAuthorityHost(options?.AuthorityHost ?? AzureAuthorityHosts.GetDefault());
```

Comment:
We validation in some setters but not this one. Why?

Reply:
We do this check here as well because the `AuthorityHost` can be set directly, but also can be set via the `AZURE_AUTHORITY_HOST` environment variable. Because of the latter case we need to validate before the host is used.